src/Repositories/UserRepository.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+namespace Launcher\Mercurius\Repositories;
+
+use Illuminate\Contracts\Pagination\LengthAwarePaginator;
+
+class UserRepository
+{
+    /**
+     * Perform a search.
+     *
+     * @param string $query
+     * @param int    $limit
+     *
+     * @return LengthAwarePaginator
+     */
+    public function search(string $query, int $limit = 6): LengthAwarePaginator
+    {
+        $userFqcn = config('mercurius.models.user');
+
+        return $userFqcn::where('name', 'LIKE', '%'.$query.'%')
```

Comment:
This simplifies things a bit, and gibberish input should still return a valid paginator (haven't tested, though).

Reply:
That looks much better, TY! I will test it

Reply:
working great, TY! :)